### ISSUE TITLE
App crashes when text is tapped

### ISSUE THREAD

**Issue by sezijlema** [minvws/nl-covid19-coronacheck-app-android#25]
### Describe the bug, issue or concern

App crashes when you tap on text in the "Make QR code" view.
E.g. on "This app makes a QR code in three steps"

### To Reproduce
1. Open app
2. Click on 'Make QR code'
3. Tap on text

### Expected behavior
Nothing

### Screenshots
No screenshots possible.

**Smartphone (please complete the following information):**
 - Device: Xiaomi Mi 9 Lite
 - OS: Android 10
 - App Version: 2.0

### Additional context
java.util.NoSuchElementException: Array is empty.
	at kotlin.collections.ArraysKt___ArraysKt.first(_Arrays.kt:1013)
		at nl.rijksoverheid.ctr.design.views.HtmlTextViewWidget.dispatchPopulateAccessibilityEvent(HtmlTextView.kt:163)
			at android.view.View.sendAccessibilityEventUncheckedInternal(View.java:8061)
				at android.view.View.sendAccessibilityEventUnchecked(View.java:8038)
					at android.widget.TextView.sendAccessibilityEventUnchecked(TextView.java:12068)
						at android.view.View.sendAccessibilityEventInternal(View.java:8015)
							at android.widget.TextView.sendAccessibilityEventInternal(TextView.java:12057)
								at android.view.View.sendAccessibilityEvent(View.java:7979)
									at android.view.View.performClick(View.java:7191)
										at android.view.View.performClickInternal(View.java:7162)
											at android.view.View.access$3500(View.java:819)
												at android.view.View$PerformClick.run(View.java:27684)
													at android.os.Handler.handleCallback(Handler.java:883)
														at android.os.Handler.dispatchMessage(Handler.java:100)
															at android.os.Looper.loop(Looper.java:224)
																at android.app.ActivityThread.main(ActivityThread.java:7562)
																	at java.lang.reflect.Method.invoke(Native Method)
																		at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
																			at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
																			

### Governance
- [x] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [x] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [x] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

**Comment by JJdeGroot** [comment]
Thanks for reporting this issue.
It has been fixed in version 2.0.2, which has been submitted to the Playstore earlier today.

It was caused by a `first()` call on an array which could be zero. `firstOrNull()` should have been used. But the Kotlin vs Swift syntax confused me.

What is odd though: Xiaomi devices somehow send accessibility events, while other devices do not.

**Comment by sezijlema** [comment]
Thanks, that's good to hear.

I have the accessibility services from Twilight (blue filter) and Bitwarden (password fill) enabled. The latter doesn't really function for me but I don't think that's relevant here.

**Comment by JJdeGroot** [comment]
I did some more research into this issue:

When any kind of AccessibilityService is enabled, such as Bitwarden or Twilight, the `dispatchPopulateAccessibilityEvent` method is called. 

When text is clicked, it sends a TYPE_VIEW_CLICKED event.
When a link is clicked, it sends a TYPE_VIEW_TEXT_SELECTION_CHANGED event.

We only listen to the TYPE_VIEW_CLICKED events. But because these events are sent, even without "real" assistive technology, the code to open links is activated.

I have added an extra check for touch exploration, which is something these kind of assistive technologies are not using.
